### PR TITLE
Send seperat tree for agenda items

### PR DESCRIPTION
**Pull request by ostcar**

### CODE REVIEW

File: openslides/agenda/views.py [hunk 1]
```diff
@@ -16,6 +17,8 @@
 from django.utils.translation import ugettext as _
 from django.utils.translation import ugettext_lazy
 from reportlab.platypus import Paragraph
+from rest_framework.decorators import list_route
```
Comment (normanjaeckel) [inline openslides/agenda/views.py]: I would like to import these things from openslides.utils.rest_api.

File: openslides/agenda/views.py [hunk 2]
```diff
@@ -811,3 +814,69 @@ def get_queryset(self):
         if not self.request.user.has_perm('agenda.can_see_orga_items'):
             queryset = queryset.exclude(type__exact=Item.ORGANIZATIONAL_ITEM)
         return queryset
+
+    @list_route(methods=['get', 'put'])
+    def tree(self, request):
+        """
+        Returns or set the agenda tree.
```
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Typo: sets

File: openslides/agenda/views.py [hunk 3]
```diff
@@ -811,3 +814,69 @@ def get_queryset(self):
...
+                    for child in item_children[item.pk]]
+
+        return Response(dict(id=item.pk, children=get_children(item))
+                        for item in item_list if not item.parent)
+
+    def set_tree(self, request):
+        """
+        Sets the agenda tree.
+        """
+        tree = request.data['tree']
```
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Here we need a permission check on 'agenda.can_manage_agenda'.

File: openslides/agenda/views.py [hunk 4]
```diff
@@ -811,3 +814,69 @@ def get_queryset(self):
...
+                break
+            touched_items.add(item_pk)
+
+            Item.objects.filter(pk=item_pk).update(
+                parent_id=parent_pk,
+                weight=weight)
+        else:
+            status = "ok"
+
+        return Response({'status': status})
```
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Here we should respond a proper status code and as content {'detail': <message>}.

File: openslides/agenda/views.py (hunk 2 above)
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Please add an example tree.

File: openslides/agenda/views.py (hunk 1 above)
Comment (ostcar) [inline openslides/agenda/views.py]: I think this is exaggerated. We import other things from other third party libs, too. 

But I can live with it, so: done

File: openslides/agenda/views.py (hunk 2 above)
Comment (ostcar) [inline openslides/agenda/views.py]: done. But in the method set_tree

File: openslides/agenda/views.py (hunk 3 above)
Comment (ostcar) [inline openslides/agenda/views.py]: done

File: openslides/agenda/views.py (hunk 4 above)
Comment (ostcar) [inline openslides/agenda/views.py]: I don't see why.

Won't fix

Comment (normanjaeckel) [inline openslides/agenda/views.py]: On every other API request the framework does so. HTTP 200 means OK and HTTP 400 means Error. Additionally the framework uses a 'detail' message. So we should do this as the framework does.

**Comment by ostcar** [comment]
all done

**Comment by normanjaeckel** [comment]
Rebase and merge.

**Pull request merged**